dashboard/src/actions/profileActions.js

```diff
@@ -0,0 +1,92 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+import * as API_ROUTES from "../utils/apiConstants";
+import Cookies from "js-cookie";
+
+export const getProfileDetails = () => async (dispatch) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const username = Cookies.get("username") ? Cookies.get("username") : null;
+
+    const response = await API.get(`${API_ROUTES.GET_USER}/${username}`);
+
+    if (response.status === 200 && Object.keys(response.data).length > 0) {
+      dispatch({
+        type: TYPES.GET_USER_DETAILS,
+        payload: response?.data,
+      });
+    } else {
+      let toast = {
+        variant: "danger",
+        title: response?.data?.message,
+      };
+      dispatch({
+        type: TYPES.SHOW_TOAST,
+        payload: { ...toast },
+      });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    dispatch({ type: TYPES.NETWORK_ERROR });
+    dispatch({ type: TYPES.COMPLETED });
+  }
+};
+
+export const updateUserDetails =
+  (value, fieldName) => async (dispatch, getState) => {
+    let userDetails = { ...getState().userProfile.userDetails };
+    userDetails[fieldName] = value;
+
+    dispatch({
+      type: TYPES.UPDATE_USER_DETAILS,
+      payload: { ...userDetails },
+    });
+  };
+
+export const sendForUpdation = () => async (dispatch, getState) => {
```

Comment:
I kinda the sound like "updation" as a verb, but just `sendForUpdate` would probably be better. 😄

Reply:
Done! :grinning: